OCIF/staking/src/lib.rs

```diff
@@ -828,24 +836,43 @@ pub mod pallet {
             let current_era = Self::current_era();
             ensure!(era < current_era, Error::<T>::IncorrectEra);
 
+            //  // Check if the ensure staking_info.total > Zero::zero(), is a good way to avoid this extra read.
+            // if let Some(unregister_era) = Self::is_core_unregistering(core_id) {
+            //     ensure!(era < unregister_era, Error::<T>::NoStakeAvailable);
+            // }
             let staking_info = Self::core_stake_info(core_id, era).unwrap_or_default();
-            let reward_and_stake =
-                Self::general_era_info(era).ok_or(Error::<T>::UnknownEraReward)?;
 
-            let (_, stakers_joint_reward) =
-                Self::core_stakers_split(&staking_info, &reward_and_stake);
-            let staker_reward =
-                Perbill::from_rational(staked, staking_info.total) * stakers_joint_reward;
+            // // Just in case the core is being unregistered and the staker was not unstaked yet.
+            // // Erroring here may require the front end to handle the case where the user is trying to claim rewards
+            // // before the core is unregistered.
+            // // I'll write another option for avoiding this error down here.
+            // ensure!(
+            //     staking_info.total > Zero::zero(),
+            //     Error::<T>::NoStakeAvailable
+            // );
+
+            let mut staker_reward = Zero::zero();
+
+            if staking_info.total > Zero::zero() {
+                let reward_and_stake =
+                    Self::general_era_info(era).ok_or(Error::<T>::UnknownEraReward)?;
+
+                let (_, stakers_joint_reward) =
+                    Self::core_stakers_split(&staking_info, &reward_and_stake);
+                staker_reward =
+                    Perbill::from_rational(staked, staking_info.total) * stakers_joint_reward;
+
+                let reward_imbalance = <T as pallet::Config>::Currency::withdraw(
+                    &Self::account_id(),
+                    staker_reward,
+                    WithdrawReasons::TRANSFER,
+                    ExistenceRequirement::AllowDeath,
+                )?;
```

Comment:
I was a bit confused regarding the best way to handle claims mid core unregistration after an era change

Reply:
@arrudagates is `if staking_info.total > Zero::zero() {` okay to use as a safety check for this scenario instead of the commented out code above?